### PR TITLE
[test_monitor] small refactoring

### PR DESCRIPTION
**Pull request by andreyor**

### CODE REVIEW

File: samples/metrics_monitor/CMakeLists.txt [hunk 1]
```diff
@@ -35,11 +35,40 @@ if( PKG_LIBDRM_FOUND AND
...
+
+    link_libraries(
+        ${PKG_LIBDRM_LIBRARIES}
+        pthread 
+        rt
+    )
+
+    set( sources "" )
+
+    file( GLOB_RECURSE srcs "${CMAKE_CURRENT_SOURCE_DIR}/igtload/*.c" )
```
Comment (dvrogozh) [inline samples/metrics_monitor/CMakeLists.txt]: Why you can't use:
`file( GLOB_RECURSE test_srcs "${CMAKE_CURRENT_SOURCE_DIR}/test/*.cpp "${CMAKE_CURRENT_SOURCE_DIR}/test/*.c"" )`
below and avoid creating a new library?

File: samples/metrics_monitor/CMakeLists.txt [hunk 2]
```diff
@@ -35,11 +35,40 @@ if( PKG_LIBDRM_FOUND AND
...
+    list( APPEND sources ${srcs})
+    
+    make_library( igtload none shared )
+    make_library( igtload_static none static )
```
Comment (dvrogozh) [inline samples/metrics_monitor/CMakeLists.txt]: no. that's wrong thing to do. We don't need shared library for sure.

Comment (andreyor) [reply samples/metrics_monitor/CMakeLists.txt]: fixed

File: samples/metrics_monitor/CMakeLists.txt [hunk 3]
```diff
@@ -35,11 +35,16 @@ if( PKG_LIBDRM_FOUND AND
     PKG_PCIACCESS_FOUND AND
     BUILD_TESTS)
 
-    file( GLOB_RECURSE test_srcs "${CMAKE_CURRENT_SOURCE_DIR}/test/*.cpp" )
+    # test_monitor sample
```
Comment (dvrogozh) [inline samples/metrics_monitor/CMakeLists.txt]: drop this comment

File: samples/metrics_monitor/CMakeLists.txt [hunk 4]
```diff
@@ -35,11 +35,16 @@ if( PKG_LIBDRM_FOUND AND
...
 
-    add_executable( test_monitor ${test_srcs})
-    target_include_directories( test_monitor PRIVATE ./include ${PKG_LIBDRM_INCLUDE_DIRS})
-    target_link_libraries( test_monitor pthread rt gtest cttmetrics ${PKG_LIBDRM_LIBRARIES})
+    set( sources "" )
```
Comment (dvrogozh) [inline samples/metrics_monitor/CMakeLists.txt]: you don't need this variable actually. remove

File: samples/metrics_monitor/CMakeLists.txt [hunk 5]
```diff
@@ -35,11 +35,16 @@ if( PKG_LIBDRM_FOUND AND
...
+
+    file( GLOB_RECURSE test_srcs "${CMAKE_CURRENT_SOURCE_DIR}/test/*.cpp" "${CMAKE_CURRENT_SOURCE_DIR}/igtload/*.c" )
+    list( APPEND sources ${test_srcs})
```
Comment (dvrogozh) [inline samples/metrics_monitor/CMakeLists.txt]: you don't need sources - remove. you use only test_srcs

File: samples/metrics_monitor/CMakeLists.txt [hunk 6]
```diff
@@ -35,11 +35,16 @@ if( PKG_LIBDRM_FOUND AND
...
+
+    add_executable( test_monitor ${test_srcs} )
+    target_include_directories( test_monitor PRIVATE ./include ./igtload ${PKG_LIBDRM_INCLUDE_DIRS} )
+    target_link_libraries( test_monitor pthread rt gtest ${LIBS} ${PKG_LIBDRM_LIBRARIES} )
```
Comment (dvrogozh) [inline samples/metrics_monitor/CMakeLists.txt]: what are ${LIBS}? Please, return old version with cttmetrics - it was simpler

File: samples/metrics_monitor/CMakeLists.txt [hunk 7]
```diff
@@ -35,11 +35,16 @@ if( PKG_LIBDRM_FOUND AND
...
 
-    file( GLOB_RECURSE test_srcs "${CMAKE_CURRENT_SOURCE_DIR}/test/*.cpp" )
+    # test_monitor sample
 
-    add_executable( test_monitor ${test_srcs})
-    target_include_directories( test_monitor PRIVATE ./include ${PKG_LIBDRM_INCLUDE_DIRS})
-    target_link_libraries( test_monitor pthread rt gtest cttmetrics ${PKG_LIBDRM_LIBRARIES})
+    set( sources "" )
+
+    file( GLOB_RECURSE test_srcs "${CMAKE_CURRENT_SOURCE_DIR}/test/*.cpp" "${CMAKE_CURRENT_SOURCE_DIR}/igtload/*.c" )
```
Comment (dvrogozh) [inline samples/metrics_monitor/CMakeLists.txt]: Can you avoid creating new directory called igtload? Place all files in test, please.

File: samples/metrics_monitor/CMakeLists.txt (hunk 4 above)
Comment (andreyor) [reply samples/metrics_monitor/CMakeLists.txt]: fixed

File: samples/metrics_monitor/CMakeLists.txt (hunk 3 above)
Comment (andreyor) [reply samples/metrics_monitor/CMakeLists.txt]: fixed

File: samples/metrics_monitor/CMakeLists.txt (hunk 5 above)
Comment (andreyor) [reply samples/metrics_monitor/CMakeLists.txt]: fixed

File: samples/metrics_monitor/CMakeLists.txt (hunk 6 above)
Comment (andreyor) [reply samples/metrics_monitor/CMakeLists.txt]: fixed

File: samples/metrics_monitor/CMakeLists.txt (hunk 7 above)
Comment (andreyor) [reply samples/metrics_monitor/CMakeLists.txt]: fixed

File: samples/metrics_monitor/CMakeLists.txt [hunk 8]
```diff
@@ -35,11 +35,11 @@ if( PKG_LIBDRM_FOUND AND
     PKG_PCIACCESS_FOUND AND
     BUILD_TESTS)
 
-    file( GLOB_RECURSE test_srcs "${CMAKE_CURRENT_SOURCE_DIR}/test/*.cpp" )
+    file( GLOB_RECURSE test_srcs "${CMAKE_CURRENT_SOURCE_DIR}/test/*.cpp" "${CMAKE_CURRENT_SOURCE_DIR}/test/*.c" )
 
-    add_executable( test_monitor ${test_srcs})
-    target_include_directories( test_monitor PRIVATE ./include ${PKG_LIBDRM_INCLUDE_DIRS})
-    target_link_libraries( test_monitor pthread rt gtest cttmetrics ${PKG_LIBDRM_LIBRARIES})
+    add_executable( test_monitor ${test_srcs} )
```
Comment (dvrogozh) [inline samples/metrics_monitor/CMakeLists.txt]: please, avoid unnecessary changes. You did not functionally modified these lines, so left them intact.

File: samples/metrics_monitor/test/igt_load.h [hunk 9]
```diff
@@ -30,24 +30,17 @@
 #define __IGT_LOAD_H__
 
 #include <stdint.h>
+#include <stdbool.h>
```
Comment (dvrogozh) [inline samples/metrics_monitor/test/igt_load.h]: why you need it? very suspicious, could be implementation specific...

Comment (andreyor) [reply samples/metrics_monitor/test/igt_load.h]: Because of

/msdk/MediaSDK/samples/metrics_monitor/test/igt_load.h:41:1: error: unknown type name ‘bool’; did you mean ‘_Bool’?
 bool gem_has_bsd2(int fd);
 ^~~~
 _Bool

**Comment by andreyor** [comment]
> why build fails?

I didn't see such messages on my system:

In file included from /usr/include/libdrm/i915_drm.h:30:0,
                 from /localdisk/bb/worker/build/build_dir/repos_forked/MediaSDK/samples/metrics_monitor/test/igt_load.h:35,
                 from /localdisk/bb/worker/build/build_dir/repos_forked/MediaSDK/samples/metrics_monitor/test/igt_load.c:29:
/usr/include/libdrm/drm.h:132:2: error: unknown type name ‘size_t’
  size_t name_len;   /**< Length of name buffer */
  ^~~~~~
/usr/include/libdrm/drm.h:134:2: error: unknown type name ‘size_t’
  size_t date_len;   /**< Length of date buffer */
  ^~~~~~
/usr/include/libdrm/drm.h:136:2: error: unknown type name ‘size_t’
  size_t desc_len;   /**< Length of desc buffer */
  ^~~~~~
/usr/include/libdrm/drm.h:146:2: error: unknown type name ‘size_t’
  size_t unique_len;   /**< Length of unique */
  ^~~~~~

File: samples/metrics_monitor/CMakeLists.txt (hunk 8 above)
Comment (andreyor) [reply samples/metrics_monitor/CMakeLists.txt]: fixed

File: samples/metrics_monitor/test/igt_load.h (hunk 9 above)
Comment (andreyor) [reply samples/metrics_monitor/test/igt_load.h]: fixed: this issue and failed build were caused by include "igt_load.h" position in igt_load.c

File: samples/metrics_monitor/test/cttmetrics_gtest.cpp [hunk 10]
```diff
@@ -31,7 +31,11 @@ File Name: cttmetrics_gtest.cpp
 #include "cttmetrics.h"
 #include "cttmetrics_utils.h"
 #include "gtest/gtest.h"
-#include "igt_load.h"
+
+extern "C" {
```
Comment (dvrogozh) [inline samples/metrics_monitor/test/cttmetrics_gtest.cpp]: It's better to add this into .h files directly:
```
#ifdef __cplusplus
extern "C" {
#endif 
```

Comment (andreyor) [reply samples/metrics_monitor/test/cttmetrics_gtest.cpp]: done

Comment (dvrogozh) [reply samples/metrics_monitor/test/cttmetrics_gtest.cpp]: no, not done. Why you did that here in .cpp? I talked about #include "device_info.h" and  #include "igt_load.h". Here in  .cpp you need to completely delete extern "C" and just include files directly.
